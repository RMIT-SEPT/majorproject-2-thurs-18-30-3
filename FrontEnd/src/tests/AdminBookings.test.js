import React from 'react'

const {default: UpdateBookingDisplay} = require('../components/admin_components/UpdateBookingDisplay')
import {StaticRouter} from 'react-router'
import renderer from 'react-test-renderer'
import {render, screen, fireEvent, leftClick, queryByText} from '@testing-library/react'
import '@testing-library/jest-dom'
import '@testing-library/jest-dom/extend-expect'

const dummyBooking = {id: 1, time: '1:23', date: '2/2/21', 
                        serviceName: 'test',customerId:1, employeeId:2}

const deleteFuncMock = jest.fn();

const setup = () => {
    return render(
      <StaticRouter>
        <UpdateBookingDisplay booking={dummyBooking} deleteFunc={deleteFuncMock}/>
      </StaticRouter>
    )
}

 //snapshot test
 test('Test Page Renders', () => {
    const comp = renderer
      .create(
        <StaticRouter>
            <UpdateBookingDisplay booking={dummyBooking} deleteFunc={deleteFuncMock}/>
        </StaticRouter>
      )
      .toJSON()
    expect(comp).toMatchSnapshot()
  })

  test('Test Page Contents', () => {
    const container = setup();
    expect(screen.getByRole('main')).toHaveTextContent('Time Slot');
    expect(screen.getByRole('main')).toHaveClass('add-booking');
    expect(screen.getAllByRole('heading').length).toBe(3);
    expect(screen.getAllByRole('button').length).toBe(1);
  })

  test('Test Booking Display', () => {
    const container = setup();
    expect(screen.getAllByRole('cell')[0]).toHaveTextContent(dummyBooking.time);
    expect(screen.getAllByRole('cell')[0]).toHaveTextContent(dummyBooking.date);
    expect(screen.getAllByRole('cell')[1]).toHaveTextContent(dummyBooking.customerId);
    expect(screen.getAllByRole('cell')[2]).toHaveTextContent(dummyBooking.employeeId);
  })

  test('Test Button', () => {
    const container = setup();
    const btn = screen.getByRole('button');
    fireEvent.click(btn);
    expect(deleteFuncMock).toHaveBeenCalled();
  })
  
  