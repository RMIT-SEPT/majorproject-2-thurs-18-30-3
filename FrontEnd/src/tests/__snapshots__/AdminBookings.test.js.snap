// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Page Renders 1`] = `
<main
  className="add-booking"
  role="main"
>
  <div
    className="add-booking-heading"
  >
    <h1>
      Time Slot
    </h1>
  </div>
  <div
    className="add-booking-body"
    role="cell"
  >
    <span>
      1:23
    </span>
    <span>
      2/2/21
    </span>
  </div>
  <hr />
  <div
    className="add-booking-body"
    role="cell"
  >
    <h4>
      Booked by:
    </h4>
    <span>
      Customer No.1
    </span>
  </div>
  <div
    className="add-booking-body"
    role="cell"
  >
    <h4>
      Employee:
    </h4>
    <span>
      Employee No. 
      2
    </span>
  </div>
  <hr />
  <button
    className="actButton"
    onClick={[MockFunction]}
  >
    DELETE
  </button>
</main>
`;
